source/vulkan/runtime_vk.cpp

```diff
@@ -398,20 +419,40 @@ void reshade::vulkan::runtime_vk::on_reset()
 	for (VkDeviceMemory allocation : _allocations)
 		vk.FreeMemory(_device, allocation, nullptr);
 	_allocations.clear();
+
+	_is_multisampling_enabled = false;
 }
 
-void reshade::vulkan::runtime_vk::on_present(uint32_t swapchain_image_index)
+void reshade::vulkan::runtime_vk::on_present(uint32_t swapchain_image_index, draw_call_tracker &tracker)
 {
 	if (!_is_initialized)
 		return;
 
+	_vertices = tracker.total_vertices();
+	_drawcalls = tracker.total_drawcalls();
+	_current_tracker = &tracker;
+
 	_swap_index = swapchain_image_index;
 
 	vk.QueueWaitIdle(_current_queue); // TOOD
 	vk.ResetCommandPool(_device, _cmd_pool[swapchain_image_index], 0);
 
 	update_and_render_effects();
+
+#if RESHADE_VULKAN_CAPTURE_DEPTH_BUFFERS
+	detect_depth_source(tracker);
+#endif
+
 	runtime::on_present();
+
+	clear_DSV_iter = 1;
+	_current_tracker->reset();
+}
+
+void reshade::vulkan::runtime_vk::on_create_graphics_pipelines(const VkGraphicsPipelineCreateInfo* pCreateInfos)
+{
+	if (pCreateInfos->sType == VK_STRUCTURE_TYPE_PIPELINE_MULTISAMPLE_STATE_CREATE_INFO)
+		_is_multisampling_enabled = true;
```

Comment:
This is not doing what you want. Just because a single pipeline has MSAA active, doesn't mean all depth-stencils are multisampled. Better to just ignore this for now (also avoids having to hook `vkCreateGraphicsPipelines`, which is a bonus).

Reply:
OK. I removed it.